services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationArea.kt

```diff
@@ -1,14 +1,43 @@
 package fi.hel.haitaton.hanke.application
 
+import fi.hel.haitaton.hanke.domain.TyomaaTyyppi
+import fi.hel.haitaton.hanke.tormaystarkastelu.AutoliikenteenKaistavaikutustenPituus
+import fi.hel.haitaton.hanke.tormaystarkastelu.Meluhaitta
+import fi.hel.haitaton.hanke.tormaystarkastelu.Polyhaitta
+import fi.hel.haitaton.hanke.tormaystarkastelu.Tarinahaitta
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulos
+import fi.hel.haitaton.hanke.tormaystarkastelu.VaikutusAutoliikenteenKaistamaariin
 import org.geojson.Polygon
 
 sealed interface ApplicationArea {
     val name: String
-    val geometry: Polygon
+
+    fun geometries(): List<Polygon>
+}
+
+data class CableReportApplicationArea(override val name: String, val geometry: Polygon) :
+    ApplicationArea {
+    override fun geometries(): List<Polygon> = listOf(geometry)
 }
 
-data class CableReportApplicationArea(override val name: String, override val geometry: Polygon) :
-    ApplicationArea
+data class ExcavationNotificationArea(
+    override val name: String,
+    val hankealueId: Int,
+    val tyoalueet: List<Tyoalue>,
+    val katuosoite: String,
+    val tyonTarkoitukset: Set<TyomaaTyyppi>,
+    val meluhaitta: Meluhaitta,
+    val polyhaitta: Polyhaitta,
+    val tarinahaitta: Tarinahaitta,
+    val kaistahaitta: VaikutusAutoliikenteenKaistamaariin,
+    val kaistahaittojenPituus: AutoliikenteenKaistavaikutustenPituus,
+    val lisatiedot: String,
+) : ApplicationArea {
+    override fun geometries(): List<Polygon> = tyoalueet.map { it.geometry }
+}
 
-data class ExcavationNotificationArea(override val name: String, override val geometry: Polygon) :
-    ApplicationArea
+data class Tyoalue(
+    val geometry: Polygon,
+    val area: Double,
+    val tormaystarkasteluTulos: TormaystarkasteluTulos,
```

Comment:
This probably shouldn't be required for requests. The UI shouldn't need to enter dummy values when creating or updating the area. It also shouldn't be able to overwrite an existing result with an arbitrary one.

The 'correct' solution would be to have a separate class for the requests, but that might create a lot of complexity here?

Reply:
There's a separate ticket for calculating the tormaystarkastelutulos for the areas in the excavation notification (https://helsinkisolutionoffice.atlassian.net/browse/HAI-2315), so someof this could solved there as well.

Reply:
You're right. Thanks for noticing this aspect. I did not think so much about the fact that these data structures are now also used in requests and responses. I did try to create separate classes for requests now but it became too large and complex change for this so I only changed `tormaystarkasteluTulos` and `lisatiedot` as nullable for now.